src/Tool/TokenStorageActorProvider.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Stof\DoctrineExtensionsBundle\Tool;
+
+use Gedmo\Tool\ActorProviderInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+
+/**
+ * Provides an actor for the extensions using the token storage.
+ */
+class TokenStorageActorProvider implements ActorProviderInterface
+{
+    private ?TokenStorageInterface $tokenStorage;
+    private ?AuthorizationCheckerInterface $authorizationChecker;
+
+    public function __construct(?TokenStorageInterface $tokenStorage = null, ?AuthorizationCheckerInterface $authorizationChecker = null)
+    {
+        $this->tokenStorage = $tokenStorage;
+        $this->authorizationChecker = $authorizationChecker;
+    }
+
+    public function getActor(): ?TokenInterface
+    {
+        if (null === $this->tokenStorage || null === $this->authorizationChecker) {
+            return null;
+        }
+
+        $token = $this->tokenStorage->getToken();
+
+        if (null === $token || !$this->authorizationChecker->isGranted('IS_AUTHENTICATED_REMEMBERED')) {
+            return null;
+        }
+
+        return $token;
```

Comment:
```suggestion
        return $token->getUser();
```

Shouldn't we return the user instead of the security token?

Reply:
we should, indeed

Reply:
Functionally it shouldn't matter.  The extensions are looking for an object with either a `getUserIdentifier()`, `getUsername()`, or `__toString()` method and `TokenInterface::getUserIdentifier()` does exist.  The difference then boils down to whether the token and user objects have different identifiers.  (FWIW the blame listener is passing the user and the loggable listener is passing the token, so one of them is "wrong" in some way)

Reply:
it does not matter when storing the value in a text field using `getUserIdentifier`. But it matters when storing it in an association.

Reply:
D'oh, right, I forgot blameable supports relations 🤦   It's updated